nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -275,6 +275,44 @@ def perturb(self, data):
         data._samples = y_stretch
 
 
+class SilencePerturbation(Perturbation):
+    """
+    Applies random silence at the start or end of the audio.
+
+    Args:
+        min_start_silence_secs (float): Min start silence level in secs
+        max_start_silence_secs (float): Max start silence level in secs
+        min_end_silence_secs (float): Min end silence level in secs
+        max_end_silence_secs (float): Max end silence level in secs
+        rng: Random number generator
+    """
+
+    def __init__(
+        self,
+        min_start_silence_secs=0,
+        max_start_silence_secs=0,
+        min_end_silence_secs=0,
+        max_end_silence_secs=0,
+        rng=None,
+        value=0,
```

Comment:
You'll need to import the libraries as well for typing, so probably best to do this in your branch and resolve!
```suggestion
        min_start_silence_secs: int=0,
        max_start_silence_secs: int=0,
        min_end_silence_secs: int=0,
        max_end_silence_secs: int=0,
        rng: Optional[Any]=None,
        value: float=0,
```